src/Term/AliasGroupFallback.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Wikibase\DataModel\Term;
+
+use InvalidArgumentException;
+
+/**
+ * Ordered set of aliases resulting from language fall back.
+ * Immutable value object.
+ *
+ * Duplicates and whitespace only values are removed. Values are trimmed.
+ *
+ * @since 2.3.0
+ *
+ * @licence GNU GPL v2+
+ * @author Jan Zerebecki < jan.wikimedia@zerebecki.de >
+ */
+class AliasGroupFallback extends AliasGroup {
+
+	private $actualLanguageCode;
+	private $sourceLanguageCode;
+
+	/**
+	 * @param string $languageCode
+	 * @param string[] $aliases
+	 * @param string $actualLanguageCode fallen back to
+	 * @param string|null $sourceLanguageCode for transliteration
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	public function __construct( $languageCode, array $aliases, $actualLanguageCode, $sourceLanguageCode ) {
```

Comment:
So *all* aliases come from the same fallback? That doesn't make sense to me. The application should be free to choose how fallabck is applied to aliases - e.g. all aliases from all applicable fallback languages could be collected. In that case, the the actual language and source language can be different for each alias.